### PR TITLE
Update TeslaFi importer to accept newer export formats

### PR DESCRIPTION
**Pull request by ithinuel**
I am not sure if using "1" as the default id but that's what's suggested in https://github.com/teslamate-org/teslamate/issues/3158#issuecomment-1721240983 and has worked so far.

### CODE REVIEW

**Comment by ithinuel** [comment]
Is the ci failure related to the changes in this PR?

**Comment by JakobLichterfeld** [comment]
> Is the ci failure related to the changes in this PR?

As far as I can see: no.

File: lib/teslamate/import/line_parser.ex [hunk 1]
```diff
@@ -76,6 +76,9 @@ defmodule TeslaMate.Import.LineParser do
       {"id", _val} ->
         Map.put(acc, "id", :rand.uniform(65536))
 
+      {"vehicle_id", ""} ->
+        Map.put(acc, "vehicle_id", "1")
```
Comment (JakobLichterfeld) [inline lib/teslamate/import/line_parser.ex:80]: hard coding vehicle id will not be merged

**Review by JakobLichterfeld — CHANGES_REQUESTED** [review]

File: lib/teslamate/import/line_parser.ex (hunk 1 above)
Comment (ithinuel) [reply lib/teslamate/import/line_parser.ex:80]: Is there any acceptable alternative? This is used in lieu of a default value.

As mentioned in the description, this what is recommended/suggested in the linked comment.

Comment (JakobLichterfeld) [reply lib/teslamate/import/line_parser.ex:80]: Proper format input file with correct vehicle id, as suggested in the linked comment. Hardcoding vehicle id to import to 1 will result in overwriting old data if that vehicle id exists before.

Comment (ithinuel) [reply lib/teslamate/import/line_parser.ex:80]: > Proper format input file with correct vehicle id, as suggested in the linked comment. 

Isn’t a "proper format input file" for this TeslaFi Importer what is directly exported from TeslaFi without manual modifications?

> Hardcoding vehicle id to import to 1 will result in overwriting old data if that vehicle id exists before.

Doesn’t the current code ignores any input data if there is already some dated before or on the day of the current entry/event?
At worse, using 1 as the default value would add data to the wrong vehicle, but should not overwrite anything.

Comment (ithinuel) [reply lib/teslamate/import/line_parser.ex:80]: @JakobLichterfeld should this use `:rand.uniform(65536)` like the `id` field?

Comment (JakobLichterfeld) [reply lib/teslamate/import/line_parser.ex:80]: No, as the vehicle id is given by tesla

Comment (ithinuel) [reply lib/teslamate/import/line_parser.ex:80]: I don’t know about tesla’s API, but the TeslaFi exports do not report it.

**Comment by ithinuel** [comment]
It is not stale.

To summarize the pending thread:

- @JakobLichterfeld does not want to see a hard codded value for the `vehicle_id`
- This `vehicle_id` is not exported in `TeslaFi`’s data exports.
- In the context of the TeslaFi importer, the input data come from TeslaFi so whether the `vehicle_id` is present or not in Tesla’s own api is irrelevant
- As far as I understand, Teslamate relies on the JSON API’s id, not the `vehicle_id`.

So, is this field essential for the purpose of TeslaMate? Could it be made optionnal and left empty by the importer?

**Comment by ithinuel** [comment]
Still waiting far input from reviewer.

File: lib/teslamate/import/line_parser.ex (hunk 1 above)
Comment (jlestel) [reply lib/teslamate/import/line_parser.ex:80]: @ithinuel Having imported many TeslaFi's, I confirm that it is tedious to fill in the vehicle_id column in each file.

This vehicle_id corresponds to the car_id of the Teslamate "cars" table.

@JakobLichterfeld Is it OK to use a new environment variable to specify the car ID for which TeslaFi data will be imported?

I should ends with something like that: 
```
      {"vehicle_id", ""} ->
        Map.put(acc, "vehicle_id", System.get_env("TESLAFI_IMPORT_VEHICLE_ID", "1"))
```

Comment (ithinuel) [reply lib/teslamate/import/line_parser.ex]: @jlestel  I (finally) applied your suggestion, and rebased on master.

It looks like CI is happy so the PR is ready for review again :)

**Review by jlestel — APPROVED** [review]
LGTM, it's much faster than adding the value in x large csv files by hand...

It might be useful to mention the `TESLAFI_IMPORT_VEHICLE_ID` in the documentation to show that we can use this env variable to choose the vehicle id on which to import the data (may be in the Requirements section of this file `website/docs/import/teslafi.md`)

**Review by JakobLichterfeld — CHANGES_REQUESTED** [review]
Thanks. Lgtm. Please also add a description about the env file in the doc, under Environment Variables and additional mention it on the doc page for tesla fi import

File: website/docs/configuration/environment_variables.md [hunk 2]
```diff
@@ -37,3 +37,4 @@ TeslaMate accepts the following environment variables for runtime configuration:
...
 | **TZ**                            | Used to establish the local time zone, e.g. to use the local time in logs. See [List of tz database time zones](https://en.wikipedia.org/wiki/List_of_tz_database_time_zones).                                                                                                                                   |                               |
 | **DEFAULT_GEOFENCE**              | The default GEOFENCE to send via GEOFENCE if car not in geofence. Overrides the default of "" which will delete any retained value.                                                                                                                                                                              | "" (no quotes)                |
+| **TESLAFI_IMPORT_VEHICLE_ID**     | The default Vehicle ID used when importing from TeslaFi. If not defined and the import source is missing that datum, 1 will be used.
```
Comment (JakobLichterfeld) [inline website/docs/configuration/environment_variables.md:40]: datum -> If it is not defined and the import source does not contain the data, 1 is used.

**Review by JakobLichterfeld — CHANGES_REQUESTED** [review]

**Review by JakobLichterfeld — APPROVED** [review]

**Comment by JakobLichterfeld** [comment]
Thank you @ithinuel!

**Pull request merged**